zenip-42202.md

```diff
@@ -0,0 +1,121 @@
+
+These non-ceasable sidechains must satisfy the following constraints:
+
+- **Sidechain Creation**
+
+  - `withdrawalEpochLength` must be set to 0 (to disable the fixed-length epochs and submission windows, otherwise they work exactly as sidechains v1 do).
+
+  - `mainchainBackwardTransferRequestDataLength` must be set to 0 (to disable the MBTR).
+
+  - `wCeasedVk` must be empty (to disable the CSW).
+
+- **Certificates**
+
+  - The `quality` field must always be set to “0”.
+
+  - If mainchain receives first a certificate of epoch N (either in mempool or in a block) and then another certificate of same epoch, the latter is rejected.
+
+  - All certificates must refer (through the `endEpochCumScTxCommTreeRoot` field) to different ordered block. So, for instance, if the mainchain receives a certificate for epoch N referring to the mainchain block M (i.e. setting the `endEpochCumScTxCommTreeRoot `field to the same value specified in the header of the mainchain block M), any certificate coming next must refer to a `endEpochCumScTxCommTreeRoot` (and thus to a block) with higher height than M (M + 1, M + 2, etc., are all good candidates). Such constraint enforces that a certificate of higher epoch refers to a more updated state of the mainchain and of the sidechain.
+
+  - More than one certificate for the same sidechain can be included in the same block (if they respect all the constraints above and are ordered sequentially).
+
+When sidechains v2 are created with `withdrawalEpochLength` not set to 0, they will enable the default ceasing mechanism as it is intended for sidechains v1.
+
+### Maturity
+
+For sidechain v0 and v1, Zendoo implements a maturity mechanism for the amount of coins transferred to and from any sidechain.
+
+For instance, whenever transactions are created with a cross-chain output (sidechain creation, forward transfer, or mainchain backward transfer request), the coins and fees are able to move to the sidechain but the amount is not immediately added to its balance. The coins are marked as immature and after some block confirmation, they are included in the sidechain balance. The number of confirmations is set to `10` blocks for mainnet in the chain parameters.
+How is this concept of maturity handled by the mainchain? Whenever a transaction including a CSW or a certificate with BTs is received, the validation rules enforce that the total amount of coins moved back from sidechain to mainchain does not exceed the sidechain balance. If a sidechain tries to move back more coins than its balance, the transaction is rejected.
+
+
+The concept of maturity also applies to backward transfers in a certificate. During the submission window phase, a certificate can be superseded by another certificate of higher quality. Because of this backward transfers included in a certificate are immature until the certificate is final.
+
```

Comment:
Why is this mechanism needed for ceasable sidechains, and not for non-ceasable ones?

Reply:
This is due to the quality mechanism: since ceasable sidechains can publish several certificates with increasing quality for a given epoch, we need to wait until the last one is published until we agree on the final state of the sidechain. Moreover, in case of ceasing, the last certificate is "discarded" and the sidechain is restored to previous state, so, to consider as "final" a certificate of epoch E, we need to wait until the end of the submission window for epoch E+1.

Consider also that for non-ceasable sidechains there is no fixed epoch length and there is no submission window.

Reply:
Ok, so if I understand this correctly, for Non-ceasable sidechains, as there is no limit (epoch or submission window) to receive a certificate from a sidechain, the only requirement is that the mainchain block referenced in that certificate is higher than the previous certificate block, so sending a second certificate for the same epoch and higher quality would break this rule. 
But then what is the benefit the quality mechanism provides for ceasable sidechains? And why is this benefit not provided for non-ceasable ones?

Reply:
As far as I know, there are at least two advantages when talking in general about the [Latus design](https://arxiv.org/pdf/2103.13754.pdf):
- in case of sidechain forks happening during the submission window, a part of the network might agree on a state S1 and submit a certificate C1 while another part might agree on S2 and send the certificate C2. The mainchain has no way to detect which state (and certificate) is legit, so keeping the quality of certificates proportional to the length of the chain it refers to, makes the mainchain been able to choose the best certificate (higher quality => certificate created from the state of the longest chain)
- if a malicious actor partially takes control of a sidechain, it can forge a sidechain block to get the longest chain (without publishing it to other peers) and then submit a certificate with highest quality. In this way, other honest peers wouldn't be able to continue forging new blocks as they don't know the block forged by the attacker and the state is already commited to mainchain. To prevent such attacks, quality is proportional to the number of nodes that signed the certificate: having more signers means that the quality is higher and that the certificate has higher chances to be considered as final by the mainchain (reducing the possibility for attackers to break the sidechain).

I'm not sure if both behaviors are currently taken into account in Blaze SDK and, if this is the case, if there is any difference between the design described in the paper for Latus and the current implementation.

Reply:
I see, and the reason why we consider this not to be a problem in non-ceasable sidechains is that we consider these ones to be used only in private/centralized environments mainly, right?

Reply:
Yes, at least for the malicious scenario.

For the possibility of forks, we don't have such issue anymore since we disabled the submission window, so the sidechain can take its time to generate the certificate and publish it only when it's sure that the state being committed is final.